lib/ansiblelint/__main__.py

```diff
@@ -97,20 +98,21 @@ def report_outcome(matches: List["MatchError"], options) -> int:
 warn_list:  # or 'skip_list' to silence them completely
 """
 
-    matched_rules = {match.rule.id: match.rule.shortdesc for match in matches}
-    for id in sorted(matched_rules.keys()):
-        if id not in options.warn_list:
-            msg += f"  - '{id}'  # {matched_rules[id]}\n"
+    matched_rules = OrderedDict({match.rule.id: match.rule for match in matches})
+    for id, rule in matched_rules.items():
+        if set([id, *rule.tags]).isdisjoint(options.warn_list):
```

Comment:
No need to make a throw-away in-memory list here:
```suggestion
        if {id, *rule.tags}.isdisjoint(options.warn_list):
```